sway-core/src/type_engine/type_info.rs

```diff
@@ -619,6 +619,10 @@ impl TypeInfo {
         }
     }
 
+    pub fn is_copy_type(&self) -> bool {
+        matches!(self, TypeInfo::Boolean | TypeInfo::UnsignedInteger(_))
```

Comment:
The version in `ResolvedType` also has `Byte` and `Unit`.  Not sure if you should call `is_unit()` here... probably, as AFAIK we don't actually use tuple/structs to represent it.

Are you actually calling this method though?  I see in other places you're using `ResolvedType::is_copy_type()`.  When it's declared `pub` like this the compiler won't tell you if it's never called.

Reply:
> The version in `ResolvedType` also has `Byte` and `Unit`. Not sure if you should call `is_unit()` here... probably, as AFAIK we don't actually use tuple/structs to represent it.
> 
> Are you actually calling this method though? I see in other places you're using `ResolvedType::is_copy_type()`. When it's declared `pub` like this the compiler won't tell you if it's never called.

I'm using this in `intrinsic_function.rs`, i.e., during type checking, where I can't use `ResolvedType::is_copy_type()`. 

Let me check if I should call `is_unit()`.